QUARKUS-5654.md

```diff
@@ -0,0 +1,34 @@
+# QUARKUS-5654 Update MP REST Client to 4.0
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-5654
+
+Upstream PRs:
+- https://github.com/quarkusio/quarkus/pull/43959
+
+## Scope of the testing
+- New overloaded `baseUri(String uri)` method so users don’t have to call URI.create()
+- New `header(String name, Object value)` method for adding headers to Client instances
+- Clarify specification CDI support (`@RestClient` qualifier is not optional)
+- Add multi-part processing example to specification.
+- Clarify the specification that when a client injected as a CDI bean falls out of scope, the client is closed
+
+## Getting familiar with the feature
+Getting familiar with
+- https://download.eclipse.org/microprofile/microprofile-rest-client-4.0/microprofile-rest-client-spec-4.0.html
```

Comment:
Alright, but shouldn't Quarkusy docs say that MP Rest client is based on the 4.0? I can't see it in https://github.com/quarkusio/quarkus/pull/43959, possibly they documented it in follow-up? It is nice that this is updated, but if I were starting with 3.21, I 'd not go through old migration guides as ... what is the point. I presume users will find out based on dependency version when inspecting MP Rest client interfaces.

Reply:
I just checked, and it looks like Quarkus team doesn't care much about MicroProfile ([1][2]). I doubt many  users care about it either.

[1] the last blog post tagged with MicroProfile is from 2013: https://quarkus.io/blog/tag/microprofile/
[2] the last version of MP is 7.0. The last one Quarkus supports is 4.1: https://microprofile.io/compatible/4-1/#overview